repl/src/repl.rs

```diff
@@ -362,17 +363,41 @@ fn set_globals(
             Ok(())
         }
         Pattern::Record { ref fields, .. } => {
-            let iter = fields
-                .iter()
-                .zip(::vm::dynamic::field_iter(&value, typ, vm));
-            for (field, (field_value, field_type)) in iter {
-                match field.value {
-                    Some(ref field_pattern) => {
-                        set_globals(vm, field_pattern, &field_type, &field_value)?
+            // FIXME: not sure if this is necessary, or if this has already
+            // been done. ::vm::dynamic::field_iter did this
+            // so I'm doing it too
+            let resolved_type = resolve::remove_aliases_cow(&*vm.global_env().get_env(), typ);
```

Comment:
👍 This is necessary. Since `type` could be an alias `MyRecord` we need to remove it to look at the type inside the alias. 

```
type MyRecord = { x : Int }
```